WooCommerce/src/androidTest/kotlin/com/woocommerce/android/e2e/screens/orders/SingleOrderScreen.kt

```diff
@@ -41,29 +42,73 @@ class SingleOrderScreen : Screen {
         ).check(ViewAssertions.matches(isDisplayed()))
     }
 
-    fun assertSingleOrderScreenWithProduct(order: OrderData): SingleOrderScreen {
+    private fun assertOrderStatus(orderStatus: String?): SingleOrderScreen {
+        assertIdAndTextDisplayed(ORDER_STATUS_TAG, orderStatus)
+        return this
+    }
+
+    private fun assertCustomerName(customerName: String): SingleOrderScreen {
+        assertIdAndTextDisplayed(ORDER_STATUS_CUSTOMER, customerName)
+        return this
+    }
+
+    private fun assertOrderId(orderId: Int): SingleOrderScreen {
         Espresso.onView(withId(TOOLBAR))
-            .check(ViewAssertions.matches(hasDescendant(withText("Order #${order.id}"))))
+            .check(ViewAssertions.matches(hasDescendant(withText("Order #$orderId"))))
             .check(ViewAssertions.matches(isDisplayed()))
+        return this
+    }
 
-        Espresso.onView(withText(order.productName))
+    private fun assertCustomerNote(customerNote: String): SingleOrderScreen {
+        Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.isDescendantOfA(withId(CUSTOMER_NOTE)),
+                Matchers.allOf(
+                    withId(R.id.notEmptyLabel),
+                    withText(customerNote)
+                )
+            )
+        )
+            .perform(NestedScrollViewExtension())
             .check(ViewAssertions.matches(isDisplayed()))
 
-        assertIdAndTextDisplayed(ORDER_STATUS_TAG, order.status)
-        // The element for customer name has a trailing space in the text value,
-        // adjusting here to match that
-        assertIdAndTextDisplayed(ORDER_STATUS_CUSTOMER, "${order.customerName} ")
+        return this
+    }
 
+    private fun assertPayments(order: OrderData): SingleOrderScreen {
         Espresso.onView(withId(AMOUNT_TOTAL))
             .perform(NestedScrollViewExtension())
-        assertIdAndTextDisplayed(AMOUNT_SHIPPING, order.shippingAmount)
-        assertIdAndTextDisplayed(AMOUNT_FEE, order.feeAmount)
+
+        if (order.productsTotalRaw.isNotBlank()) {
+            assertIdAndTextDisplayed(AMOUNT_PRODUCTS_TOTAL, order.productsTotalAmount)
+        }
+
+        if (order.shippingRaw.isNotBlank()) {
+            assertIdAndTextDisplayed(AMOUNT_SHIPPING, order.shippingAmount)
+        }
+
+        if (order.feeRaw.isNotBlank()) {
+            assertIdAndTextDisplayed(AMOUNT_FEE, order.feeAmount)
+        }
+
+        if (order.taxesRaw.isNotBlank()) {
+            assertIdAndTextDisplayed(AMOUNT_TAXES, order.taxesAmount)
+        }
+
         assertIdAndTextDisplayed(AMOUNT_TOTAL, order.total)
```

Comment:
what do you think about using `with` so that `order` is not repeated on each of the if condition:

```suggestion
        with(order) {
            if (productsTotalRaw.isNotBlank()) {
                assertIdAndTextDisplayed(AMOUNT_PRODUCTS_TOTAL, productsTotalAmount)
            }

            if (shippingRaw.isNotBlank()) {
                assertIdAndTextDisplayed(AMOUNT_SHIPPING, shippingAmount)
            }

            if (feeRaw.isNotBlank()) {
                assertIdAndTextDisplayed(AMOUNT_FEE, feeAmount)
            }

            if (taxesRaw.isNotBlank()) {
                assertIdAndTextDisplayed(AMOUNT_TAXES, taxesAmount)
            }

            assertIdAndTextDisplayed(AMOUNT_TOTAL, total)
        }
```

Reply:
Thanks for the suggestion, Jos!

Even though it adds two more lines and another level of indentation, removing 8 occurrences of `order` justifies this. Done with bb7e44475ae21fa6be8f93cd62545f4a93dcb4a7 (I did not commit your suggestion because of the missing line break above `with(order) {` 🙃).